File: src/creational/abstract-factory/python/classes/factories/enterprise_vehicle_factory.py
from classes.enterprise_vehicle import EnterpriseVehicle
from classes.enterpriser import Enterpriser
from classes.interfaces.user_protocol import UserProtocol
from classes.interfaces.user_vehicle_factory_protocol import \
    UserVehicleFactoryProtocol
from classes.interfaces.vehicle_protocol import VehicleProtocol


class EnterpriseVehicleFactory(UserVehicleFactoryProtocol):
    @staticmethod
    def create_user(name: str, location: str) -> UserProtocol:
        return Enterpriser(name, location)

    @staticmethod
    def create_vehicle(name: str) -> VehicleProtocol:
        return EnterpriseVehicle(name)

File: src/creational/abstract-factory/typescript/classes/factories/EnterpriseVehicleFactory.ts
import { Enterpriser } from '../Enterpriser';
import { EnterpriseVehicle } from '../EnterpriseVehicle';
import { UserProtocol } from '../interfaces/UserProtocol';
import { UserVehicleFactoryProtocol } from '../interfaces/UserVehicleFactoryProtocol';
import { VehicleProtocol } from '../interfaces/VehicleProtocol';

export class EnterpriseVehicleFactory implements UserVehicleFactoryProtocol {
  createUser(name: string, location: string): UserProtocol {
    return new Enterpriser(name, location);
  }
  createVehicle(name: string): VehicleProtocol {
    return new EnterpriseVehicle(name);
  }
}
